grakn-core/src/main/java/ai/grakn/Keyspace.java

```diff
@@ -49,6 +49,7 @@
 
     @Override
     public int compareTo(Keyspace o) {
+        if(equals(o)) return 0;
```

Comment:
Did this make Sonarcloud sad also?

Reply:
Yep. The `compareTo` was not obeying the equality contract

Reply:
Oh, [this one](https://sonarcloud.io/project/issues?id=ai.grakn%3Agrakn&issues=AWFnFYlYr0KnSpTG8FT6&open=AWFnFYlYr0KnSpTG8FT6)?

Findbugs got upset about the same thing. I don't _think_ your change will make it happy. It's upset because we have implemented `compareTo` but we haven't implemented `equals`. This is because we expect `@AutoValue` to implement `equals` for us.

For Findbugs I just disabled that check for certain classes... I don't know what we can do here though.

Reply:
Random thought... you might be able to fix the issue by explicitly declaring `public abstract boolean equals(...) {` and `public abstract int hashCode() {` in `Keyspace`. That way findbugs/sonar will understand that subclasses _must_ provide an `equals` and `hashCode` implementation.